core/src/main/java/hudson/Util.java

```diff
@@ -1927,4 +1927,23 @@ public static long daysElapsedSince(@NonNull Date date) {
     private static PathRemover newPathRemover(@NonNull PathRemover.PathChecker pathChecker) {
         return PathRemover.newFilteredRobustRemover(pathChecker, DELETION_RETRIES, GC_AFTER_FAILED_DELETE, WAIT_BETWEEN_DELETION_RETRIES);
     }
+
+    /**
+     * Returns SHA-256 Digest of input bytes
+     */
+    @Restricted(NoExternalUse.class)
+    public static byte[] getSHA256DigestOf(@NonNull byte[] input)
+    {
+        try {
+            if (input != null) {
```

Comment:
input is marked as `NonNull` - either it is `NonNull` and this check is incorrect , or it is nullable.

if it is nullable - I would say this check should come before the try